src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
+    try:
+        build_request = QueueBuildRequest(
+            image_name=image_name,
+            source_location=source_location,
+            build_parameters=build_parameters,
+            is_push_enabled=is_push_enabled,
+            timeout=timeout,
+            platform=platform,
+            build_arguments=build_arguments)
+
+        result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+        print("Successfully queued a build with build-id: {}.".format(result.build_id))
+
+        if show_logs == 'true':
+            print("Starting to stream the logs...")
+            return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    except Exception as err:
+        raise CLIError(err)
+
+
+def _check_remote_source_code(source_location):
+
+    # NOTE: acr_build doesn't support git repo url
+    if source_location.lower().startswith("git://"):
+        return source_location
+
+
+    response = requests.head(source_location)
+    if response.status_code < 400:
+        return source_location
+    else:
+        raise CLIError("'{}' doesn't exist.".format(source_location))
+
+
+def _check_image_name(image_name):
+    
+    # referenc: https://github.com/docker/distribution/tree/master/reference
+
+    if image_name == "":
+        raise CLIError("'--tag' value should not be empty.")
+    
+    tokens = image_name.split(':')
+    if(len(tokens) > 2):
+        raise CLIError("'--tag' value should be repository-name and optionally a tag in the 'repository-name:tag' format")
+
+    # check repository-name    
+    repository_name = tokens[0]
+    if len(repository_name) > 255:
+        raise CLIError("The repository-name of '--tag' value should be no more than 255 characters.")
     else:
-        # Maybe just raise error?
-        return result
+        if re.match(r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$", repository_name) is None:
+            raise CLIError("The '--tag' value is not valid. Please check https://docs.docker.com/engine/reference/commandline/tag/.")
+
+    # check tag
+    if len(tokens) == 2:
+        tag = tokens[1]
+        if re.match(r"^[\w][\w.-]{0,127}$", tag) is None:
+            raise CLIError("The '--tag' value is not valid. Please check https://docs.docker.com/engine/reference/commandline/tag/.")
+            
+    return image_name
+
+def _upload_source_code(client, registry_name, resource_group_name, source_location):
+
+    tar_file_path = os.path.join(tempfile.gettempdir(),
+                                 "source_archive_{}.tar.gz".format(hash(os.times())))
+
+    try:
+        logger.debug("Starting to archive the source code to '{}'.".format(tar_file_path))
+
+        # TODO: Check if it works in windows
+        if not source_location.endswith('/'):
```

Comment:
remove this; just use `os.path.join(path, next-thing)` for all the below calls, it will do the right thing. or you can do `os.path.sep`

Reply:
The purpose of the code is to append / so that tarfile won't add the passed-in source-location to the name which sometimes can block unpacking. I need to do some tests on windows to check the solution.

Reply:
@northtyphoon I see, I was looking at the dockerfile case and not the issue with the tar itself. You should just be able to use `os.path.sep` here and it will handle all the cases.